airflow/auth/managers/base_auth_manager.py

```diff
@@ -121,17 +126,25 @@ def get_user_id(self) -> str | None:
             return str(user_id)
         return None
 
-    @abstractmethod
+    @deprecated(
+        reason="Only used in Airflow 2 legacy UI. Will be deleted soon.",
+        category=RemovedInAirflow3Warning,
+    )
     def is_logged_in(self) -> bool:
         """Return whether the user is logged in."""
+        raise NotImplementedError()
 
     @abstractmethod
     def get_url_login(self, **kwargs) -> str:
         """Return the login page url."""
 
-    @abstractmethod
+    @deprecated(
+        reason="Only used in Airflow 2 legacy UI. Will be deleted soon.",
```

Comment:
We aren't going to support a logout url in AF3?

Reply:
I dont think so. Logging out means session. There is no session in AF3, we are using a JWT token to authenticate and authorize. Once the user is authenticate, the JWT token still be stored in the session storage (front-end). To log out users, the front-end will just need to delete this token from the session storage.